src/main/java/frc/robot/subsystems/HandlerSubsystem.java

```diff
@@ -0,0 +1,122 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import com.revrobotics.REVLibError;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.ClosedLoopConfig.FeedbackSensor;
+import com.revrobotics.spark.config.SparkBaseConfig.IdleMode;
+import com.revrobotics.spark.config.SparkMaxConfig;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.HandlerConstants;
+
+public class HandlerSubsystem extends SubsystemBase {
+  private enum State {
+    EMPTY,
+    INTAKING,
+    LOADING,
+    LOADED,
+    EJECTING,
+    CANCELLING
+  }
+
+  private State m_state;
+  private final SparkMax m_motor;
+  private final RelativeEncoder m_encoder;
+
+  private double m_intakeSpeed = HandlerConstants.kIntakeSpeed;
+  private double m_ejectSpeed = HandlerConstants.kEjectSpeed;
+
+  public HandlerSubsystem(int driveMotorChannel) {
+    m_state = State.EMPTY; //TODO: chooser
+    m_motor = new SparkMax(driveMotorChannel, MotorType.kBrushless);
+
+    SparkMaxConfig config = new SparkMaxConfig();
+    config.inverted(HandlerConstants.kMotorReversed)
+      .smartCurrentLimit(HandlerConstants.kMotorCurrentLimit)
+      .idleMode(IdleMode.kBrake);
+    config.encoder
+      .velocityConversionFactor(HandlerConstants.kVelocityConversionFactor);
+    config.closedLoop
+      .feedbackSensor(FeedbackSensor.kPrimaryEncoder);
+
+    REVLibError configureError = m_motor.configure(config, ResetMode.kResetSafeParameters, Constants.kPersistMode);
+    if (configureError != REVLibError.kOk) {
+      throw new UncheckedIOException("Failed to configure handler motor", new IOException());
+    }
+
+    m_encoder = m_motor.getEncoder();
+
+    //TODO: sensor configs here
+  }
+
+  public void intake() {
+    if (m_state != State.LOADED) {
```

Comment:
I think we're missing some edge cases here. For example, what if the state is `LOADING`? We don't want to disrupt that in-progress loading. I suggest we use exhaustive switch statements in these public methods, so that we explicitly consider all possible state transitions.